hack/tools/extension-generator/options.go

```diff
@@ -33,6 +34,7 @@ func (o *Options) AddFlags(flags *flag.FlagSet) {
 	flags.StringVar(&o.ProviderType, "provider-type", "", "Type of the provider")
 	flags.StringArrayVar(&o.ComponentCategories, "component-category", nil, fmt.Sprintf("Category of the component, one of %v", validCategories.UnsortedList()))
 	flags.StringVar(&o.Destination, "destination", "", "The path the extension manifest is written to")
+	flags.BoolVar(&o.ExtensionInjectGardenKubeconfig, "extension-inject-garden-kubeconfig", false, "Mark ControllerDeployment to inject garden kubeconfig")
```

Comment:
Instead of adding another arg can we rather pick a default value (probably `false`) to create the example here? If an extension needs the Garden Kubeconfig, they can still overwrite this value via Kustomize in their repo (see [provider-local](https://github.com/gardener/gardener/commit/54fcfeaf49db05aa7501ec50c6cd05a2cc34ecb4)).

Unlike the `generate-controller-registration.sh` we tried to reduce the command line args by not binding them to API fields. The OCI repo args are the only exception because they are not known upfront and only calculated during the release process.

Reply:
I dropped this field entirely from the generator. Following your point, an extension requiring the inject the garden kubeconfig can set the field to `true` via kustomize (as done for `provider-local`).